extensions/chathistory.md

```diff
@@ -104,10 +104,18 @@ This is useful for retrieving conversation context around a single message.
 
 #### `BETWEEN`
 
-    CHATHISTORY BETWEEN <target> <timestamp=YYYY-MM-DDThh:mm:ss.sssZ | msgid=1234> <timestamp=YYYY-MM-DDThh:mm:ss.sssZ | msgid=1234> <limit>
+    CHATHISTORY BETWEEN <target> <timestamp=YYYY-MM-DDThh:mm:ss.sssZ | msgid=1234 | cursor=1234> <timestamp=YYYY-MM-DDThh:mm:ss.sssZ | msgid=1234> <limit>
 
 Request up to `limit` number of messages between the given `timestamp` or `msgid` values. With respect to the limit, the returned messages MUST be counted starting from and excluding the first message selector, while finishing on and excluding the second. This may be forwards or backwards in time.
 
+#### `CURSORS`
+
+All commands above MAY return a `CHATHISTORY CURSORS` message. There MUST NOT be more than one message returned. If returned, that message MUST be the first one in the batch.
+
+    CHATHISTORY CURSORS <target> <next> [prev]
+
+`next` is a a cursor which can be used to query the next page of messages. `prev` is a cursor which can be used to query the previous page of messages, it MUST only be returned in response to `CHATHISTORY AROUND`.
```

Comment:
What if a server has a `prev` cursor but no `next` cursor in response to `CHATHISTORY AROUND`?

Reply:
What about `CHATHISTORY CURSORS <target> {<next>|*} {<prev>|*}` for `CHATHISTORY AROUND` and CHATHISTORY CURSORS <target> <next>` for the others?